### PR TITLE
Add C++/WinRT WinUI3 Unit Test Project Template

### PR DESCRIPTION
**Pull request by kmahone**
This updates the vsix to add a new project template for a C++/WinRT WinUI Unit Test App. It uses CppUnitTestFramework to execute.

### CODE REVIEW

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/readme.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+========================================================================
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/readme.txt:1]: looks like C++/WinRT app project template text - change to something more unittest appropriate?

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/MainWindow.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#include "MainWindow.xaml.h"
+#if __has_include("MainWindow.g.cpp")
+#include "MainWindow.g.cpp"
+#endif
+
+using namespace winrt;
+using namespace Microsoft::UI::Xaml;
+
+// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/MainWindow.cpp:11]: >http://aka.ms/winui-project-info [](http://example.com/codeflow?start=46&length=32)

we should chat with Steve White about adding some docs for the new C#, C++ unit test templates

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/pch.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#include <winrt/Microsoft.UI.Xaml.Controls.h>
+#include <winrt/Microsoft.UI.Xaml.Controls.Primitives.h>
+#include <winrt/Microsoft.UI.Xaml.Data.h>
+#include <winrt/Microsoft.UI.Xaml.Interop.h>
+#include <winrt/Microsoft.UI.Xaml.Markup.h>
+#include <winrt/Microsoft.UI.Xaml.Media.h>
+#include <winrt/Microsoft.UI.Xaml.Navigation.h>
+#include <winrt/Microsoft.UI.Xaml.Shapes.h>
+#include <winrt/Microsoft.UI.Dispatching.h>
+#include <wil/cppwinrt_helpers.h>
```
Comment (Scottj1s) [inline dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/pch.h:25]: is this (and the nuget ref) "just in case"? don't see any use of it

**Review by Scottj1s — APPROVED** [review]
LGTM - just the readme.txt update

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/pch.h (hunk 3 above)
Comment (kmahone) [reply dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/pch.h:25]: I copied the main C++ app project and used that as a basis for this one. I also don't see it being used anywhere in the main app project, so in theory we could get rid of it.
That being said, including it in the template seems like a good way to advertise the existence of this package and encourage its use.

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/MainWindow.cpp (hunk 2 above)
Comment (kmahone) [reply dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/MainWindow.cpp:11]: Yes. I have a task tracking getting good documentation for this.

File: dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/readme.txt (hunk 1 above)
Comment (kmahone) [reply dev/VSIX/ProjectTemplates/Desktop/CppWinRT/UnitTestApp/readme.txt:1]: I've updated it. Once we get a doc page up we can update this readme to point to it.

**Pull request merged**